packages/plugin-restify/src/restify.js

```diff
@@ -0,0 +1,74 @@
+const domain = require('domain') // eslint-disable-line
+const extractRequestInfo = require('./request-info')
+const createReportFromErr = require('@bugsnag/core/lib/report-from-error')
+const handledState = {
+  severity: 'error',
+  unhandled: true,
+  severityReason: {
+    type: 'unhandledErrorMiddleware',
+    attributes: { framework: 'Restify' }
+  }
+}
+
+module.exports = {
+  name: 'restify',
+  init: client => {
+    const requestHandler = (req, res, next) => {
+      const dom = domain.create()
+
+      // Start a session whether sessions are used or not. We use this
+      // to store request-specific info, in case of any errors.
+      const sessionClient = client.startSession()
+
+      // attach it to the request
+      req.bugsnag = sessionClient
+
+      // extract request info and pass it to the relevant bugsnag properties
+      const requestInfo = extractRequestInfo(req)
+      sessionClient.metaData = { ...sessionClient.metaData, request: requestInfo }
+      sessionClient.request = {
+        clientIp: requestInfo.clientIp,
+        headers: requestInfo.headers,
+        httpMethod: requestInfo.httpMethod,
+        url: requestInfo.url,
+        referer: requestInfo.referer
+      }
+
+      // unhandled errors caused by this request
+      dom.on('error', (err) => {
+        req.bugsnag.notify(createReportFromErr(err, handledState), {}, (e, report) => {
+          if (e) return client._logger('Failed to send report to Bugsnag')
+          req.bugsnag.config.onUncaughtException(err, report, client._logger)
+        })
+        if (!res.headersSent) {
```

Comment:
why do we send a response here? It seems beyond the scope of error reporting. The Koa integration only sets the response code to 500 in this scenario

```
if (!ctx.response.headerSent) ctx.response.status = 500
```

Reply:
We've caught an error which prevented a crash. Once the error is reported the process will be killed. It makes sense to me to send an HTTP error response.

The [express plugin does](https://github.com/bugsnag/bugsnag-js/pull/384/files#diff-ac05c4d39c4020ebd563acbdc91daefdR35). Setting the status code and no body has the same effect in Koa. It will look internally for the standard HTTP error message for the provided code.